File: sdks/python/alith/agent.py
from dataclasses import dataclass, field
from typing import List, Union, Callable
from .tool import Tool, create_delegate_tool
from ._alith import DelegateAgent as _DelegateAgent
from ._alith import DelegateTool as _DelegateTool


@dataclass
class Agent:
    name: str
    model: str
    tools: List[Union[Tool, Callable, _DelegateTool]] = field(default_factory=list)

    def prompt(self, prompt: str) -> str:
        tools = [
            create_delegate_tool(tool) if isinstance(tool, Callable) else tool
            for tool in self.tools or []
        ]
        agent = _DelegateAgent(self.name, self.model, tools)
        return agent.prompt(prompt)

File: sdks/node/alith/agent.ts
import { DelegateAgent, DelegateTool } from '../index.js'

// Represents an agent that can process prompts using tools
class Agent {
  /**
   * Creates an instance of Agent.
   * @param {string} name - The name of the agent.
   * @param {string} model - The model used by the agent.
   * @param {Array<DelegateTool>} tools - A list of tools available to the agent.
   */
  private _agent: DelegateAgent
  constructor(public name: string, public model: string, public tools: Array<DelegateTool> = []) {
    this._agent = new DelegateAgent(this.name, this.model, this.tools)
  }

  /**
   * Processes a prompt using the agent's tools and model.
   * @param {string} prompt - The input prompt to process.
   * @returns {string} - The result of processing the prompt.
   */
  prompt(prompt: string): string {
    // Process the prompt and return the result
    return this._agent.prompt(prompt)
  }
}

export { Agent }
